packages/middleware-sdk-s3-control/src/process-arnables-plugin/getOutpostEndpoint.ts

```diff
@@ -14,6 +14,13 @@ export const getOutpostEndpoint = (
     return hostname;
   }
 
+  const match = hostname.match(REGEX_S3CONTROL_HOSTNAME);
+
+  if (!match) {
+    // outposts hostname was already set by endpoints ruleset.
+    return hostname;
+  }
```

Comment:
this is the primary runtime logical change in this PR.

We can either fall back to accepting the hostname as-is, or remove this plugin?